calus.js

```diff
@@ -1,228 +1,185 @@
+                }
 
-                        days.push({
-                            time: day,
-                            isToday: day.hasSame(this.now, 'day'),
-                            isAvailable: isAvailable,
-                            isPast: startOf < startOfToday,
-                            isFuture: startOf > startOfToday,
-                            isThisMonth: day.hasSame(date, 'month'),
-                        })
-                    }
+                const sameMonth = date.startOf('month').hasSame(this.now, 'month');
+                const allowScroll = !this.allowPrevScroll && sameMonth;
 
-                    months.push({
-                        time: date.startOf('month'),
-                        isCurrentMonth: +days[10].time.startOf('month') == +startOfToday.startOf('month'),
-                        isInCurrentYear: +days[10].time.startOf('year') == +startOfToday.startOf('year'),
-                        days: days
-                    })
+                months.push({
+                    time: date.startOf('month'),
+                    isCurrentMonth: +days[10].time.startOf('month') == +startOfToday.startOf('month'),
+                    isInCurrentYear: +days[10].time.startOf('year') == +startOfToday.startOf('year'),
+                    days: days,
+                    disablePrevScroll: allowScroll
+                })
 
-                    date = date.plus({ months: 1 })
-                }
+                date = date.plus({ months: 1 })
+            }
 
-                return months
+            return months
+        }
+    },
+    methods: {
+        // select a day
+        select: function (day) {
+            if (day.isAvailable) {
+                // selected state is handled by direct DOM control
+                // it's a bit hacky, but it prevents a full redraw
+                // of the calendar, making it feel a lot snappier,
+                // especially with large amount of displayed dates
+                this.resetSelected()
+                this.selectedDate = day.time
+
+                this.addSelectedStyle(this.selectedDate)
+
+                this.onDateSelect(day);
             }
         },
-        methods: {
-            // select a day
-            select: function (day) {
-                if (day.isAvailable) {
-                    // selected state is handled by direct DOM control
-                    // it's a bit hacky, but it prevents a full redraw
-                    // of the calendar, making it feel a lot snappier,
-                    // especially with large amount of displayed dates
-                    this.resetSelected()
-                    this.selected = day.time
-
-                    this.addSelectedStyle(this.selected)
-
-                    this.onSelect(day)
-                }
-            },
-            setAvailable: function (array) {
-                this.resetSelected()
+        setAvailable: function (array) {
+            this.resetSelected()
```

Comment:
As stated above, mark this as Deprecated (`console.warn()`) as handled via properties